common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -412,9 +412,9 @@ protected Collation buildCollation() {
             "UTF8_BINARY_LCASE",
             PROVIDER_SPARK,
             null,
-            UTF8String::compareLowerCase,
```

Comment:
So we will keep `compareLowerCase` in `UTF8String` and  `lowercaseCompare` in `CollationAwareUTF8String`?

Let's:
a) Make naming consistent.
b) If possible remove `UTF8String::compareLowerCase` if it is no longer needed.

Reply:
I like the idea by @dbatomic.
I think the `UTF8String` class should just have functionality for the UTF8_BINARY collation. Whatever is relevant to UTF8_BINARY_LCASE looks like it should be in a separate class. Mixing functionality in confusing for the reader of either class.